enterprise/server/webhooks/webhook_data/webhook_data.go

```diff
@@ -2,6 +2,13 @@ package webhook_data
 
 // WebhookData represents the data required to run a workflow.
 type WebhookData struct {
+	// RepoURL points to the repo containing the sources needed for the workflow.
+	//
+	// This will be different from the workflow repo if the workflow is run on a forked
+	// repo as part of a pull request.
+	//
+	// If not explicitly provided, the workflow repo URL will be used.
+	RepoURL string
```

Comment:
if we use a non-trusted RepoURL we should probably use a different namespace

Separately, can you add a TODO that we cannot add any workspace re-use features when the RepoURL is untrusted?

Reply:
Agree about using a different namespace -- done. (I took a SHA256 of the repo URL, and made sure to use the repo URL from the webhook data, which is the "canonical" URL that may be different than the one stored in our table, e.g. ".git" suffix, ssh:// protocol, etc.)

Will keep the workspace re-use stuff in mind when we start working on it -- not sure if a TODO will help here since the impl of workspace reuse will likely be more in the executor code